docs_src/design/adr/0026-Common Configuration.md

```diff
@@ -0,0 +1,401 @@
+# Common Configuration
+### Submitters
+- Lenny Goodell (Intel)
+- Elizabeth Lee (Intel)
+
+## Change Log
+- [pending](URL of PR TBD) (2022-11-29)
+
+## Referenced Use Case(s)
+- [Common Configuration UCR](URL of use case link TBD)
+
+## Context
+This ADR describes the architecture of the new common configuration capability which impacts all services. Requirements for this new capability are described in the above referenced UCR. This is deemed architecturally significant due to the cross-cutting impacts.
+
+### Current Design
+
+The following flow chart demonstrates the bootstrapping of each services' configuration in the current Levski release.
+
+![](common-config-images/EdgeX 2.x Configuration bootstrapping flowchart.png)
+
+## Proposed Design
+The configuration settings that are common to all services will be partitioned out into a separate common configuration source. This  common configuration source will be pushed into the Configuration Provider by a new `core-common-config` service. When the Configuration Provider is not used the `core-common-config` service will provide the common configuration via a RESTful endpoint.

```

Comment:
When CP isn't used, the `-cc/--commonConfig` given in the design should accept a URI to the file. This was to allow loading from a URL (`http`/`https` schemes) or a file (`file` scheme). I don't think the web service or a RESTful API are needed as a generic fallback.

```suggestion
The configuration settings that are common to all services will be partitioned out into a separate common configuration source. When the Configuration Provider is used, this common configuration source will be pushed into the Configuration Provider by a new component called `core-common-config`.
```

Reply:
Or `core-common-config-bootstrapper` to be consistent with other [such services](https://github.com/edgexfoundry/edgex-go/tree/main/cmd).

Reply:
@iain-anderson @cloudxxx8 Are you ok with removing the default common config source when not using a config provider?